L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesProducer.cc

```diff
@@ -20,25 +20,22 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "Geometry/Records/interface/MuonGeometryRecord.h"
-#include "L1Trigger/CSCCommonTrigger/interface/CSCTriggerGeometry.h"
 #include "CondFormats/DataRecord/interface/CSCBadChambersRcd.h"
 #include "Geometry/GEMGeometry/interface/GEMGeometry.h"
-#include "Geometry/RPCGeometry/interface/RPCGeometry.h"
 
-//#include "DataFormats/CSCDigi/interface/CSCComparatorDigiCollection.h"
-//#include "DataFormats/CSCDigi/interface/CSCWireDigiCollection.h"
 #include "DataFormats/CSCDigi/interface/CSCALCTDigiCollection.h"
 #include "DataFormats/CSCDigi/interface/CSCCLCTDigiCollection.h"
 #include "DataFormats/CSCDigi/interface/CSCCorrelatedLCTDigiCollection.h"
-
 #include "DataFormats/GEMDigi/interface/GEMCoPadDigiCollection.h"
 
 // Configuration via EventSetup
 #include "CondFormats/CSCObjects/interface/CSCDBL1TPParameters.h"
 #include "CondFormats/DataRecord/interface/CSCDBL1TPParametersRcd.h"
 
 
-CSCTriggerPrimitivesProducer::CSCTriggerPrimitivesProducer(const edm::ParameterSet& conf) : iev(0) {
+CSCTriggerPrimitivesProducer::CSCTriggerPrimitivesProducer(const edm::ParameterSet& conf)
+{
+  config_ = conf;
```

Comment:
hi @dildick - learning from the work Matti is doing currently - please avoid caching the configuration.  that looks like a longer term project however...

Reply:
I need it for this line: `return std::unique_ptr<CSCTriggerPrimitivesBuilder>(new CSCTriggerPrimitivesBuilder(config_));`